homeassistant/components/thermostat/homematic.py

```diff
@@ -97,9 +132,10 @@ def target_temperature(self):
 
     def set_temperature(self, temperature):
         """Set new target temperature."""
-        self.device.setValue(self._full_device_name,
-                             PROPERTY_SET_TEMPERATURE,
-                             temperature)
+        device = ServerProxy(self.address)
```

Comment:
I see you've resolved the threading issues by instantiating new connection objects for every request. This means that we could swamp the device we're hitting because too many requests are coming in. Alternative could be that you instantiate a lock in `setup_platform` that you pass to each device so that we only make 1 connection at a time.